platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/state/DefaultStateManagementComponent.java

```diff
@@ -290,9 +292,14 @@ public DefaultStateManagementComponent(
      * @param signedState the newly complete signed state
      */
     private void stateHasEnoughSignatures(final SignedState signedState) {
-        if (signedState.isStateToSave()) {
-            signedStateFileManager.saveSignedStateToDisk(signedState);
+        final StateToDiskReason reason = signedState.getStateToDiskReason();
```

Comment:
Even if you don't store a boolean value internally in the `SignedState` object, I think the method `isStateToSave()` makes this code more readable. Triggering state saving whenever reason is non-null seems like a funky API.

Reply:
done